akka-typed/src/main/scala/akka/typed/internal/ActorCell.scala

```diff
@@ -0,0 +1,433 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.typed
+package internal
+
+import akka.actor.InvalidActorNameException
+import akka.util.Helpers
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+import akka.dispatch.ExecutionContexts
+import scala.concurrent.ExecutionContextExecutor
+import akka.actor.Cancellable
+import akka.util.Unsafe.{ instance ⇒ unsafe }
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.Queue
+import scala.annotation.{ tailrec, switch }
+import scala.util.control.NonFatal
+import scala.util.control.Exception.Catcher
+import akka.event.Logging.Error
+import akka.event.Logging
+
+/**
+ * INTERNAL API
+ */
+object ActorCell {
+  /*
+   * Description of the _status field bit structure:
+   *
+   * bit 0-29: activation count (number of (system)messages)
+   * bit 30: terminating (or terminated)
+   * bit 31: terminated
+   *
+   * Activation count is a bit special:
+   * 0 means inactive
+   * 1 means active without normal messages (i.e. only system messages)
+   * N means active with N-1 normal messages (plus possibly system messages)
```

Comment:
Yes, and fairness is not something that the Actor (i.e. the perpetrator) can or should assess individually. We should definitely discuss this and implement a solution, but doing it in this PR seems the wrong place to me. I’ve added it to the list of items to be fixed, so if this is the last concern then I’ll turn that list into actual tickets, fix the docs, and hopefully get this first step merged soon—because afterwards all these issues can be worked on independently by different people.